x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/TypeResolutions.java

```diff
@@ -45,6 +46,29 @@ public static TypeResolution isIP(Expression e, String operationName, ParamOrdin
         return isType(e, dt -> dt == IP, operationName, paramOrd, "ip");
     }
 
+    public static TypeResolution isEnum(Expression e, String operationName, ParamOrdinal paramOrd, Set<String> acceptedValues) {
```

Comment:
Unless this method is used in multiple places, I would not promote it to `TypeResolution`.
The name also looks incorrect, `isEnum` but there's no enum in the args - if I'm guessing the intent correctly, pass an `Enum` along side a translating function so the object/string mapping is generalized:

`<E extends Enum<E>> isStringInEnum(Expression e, Function<Enum<E>, String> value, Enum<E> e, String operationName, ParamOrdinal paramOrd)`

This however seems too complicated for a narrow case, just checking a string inside a list of strings is much more common but that should be reflected in the name.